python/graphstorm/graphpartitioning/metadata_schema.py

```diff
@@ -0,0 +1,706 @@
+        self._global_nid_offsets = None
+        self._global_eid_offsets = None
+        self._ntypes = None
+        self._ntype_id_map = None
+        self._id_ntype_map = None
+        self._etype_id_map = None
+        self._id_etype_map = None
+        self._etypes = None
+        self._ntype_features = None
+        self._etype_features = None
+        self._ntype_feature_files = None
+        self._etype_feature_files = None
+        self._etype_files = None
+
+        self.__load_json(file_name, base_dir)
+        if init_maps:
+            self.__init_maps()
+            self.__validate_metadata()
+
+    ############################## Private Methods #############################
+
+    def __check_ifexists(self, data_files, file_type):
+        """Helper function to verify the presence of a set of filenames.
+
+        Argument
+        --------
+        data_files: list of str
+            A set of file names, either in absolute or in relative formats.
+        src_type: str
+            Either node type or edge type from the input graph.
+        """
+        for file_name in data_files:
+            file_path = file_name
+            if not os.path.isabs(file_path):
+                file_path = os.path.join(self._basedir, file_path)
+            assert os.path.isfile(
+                file_path
+            ), f"File: {file_path} does not exist for {file_type}."
+
+    def __init_maps(self):
+        """Initialization function to build all the necessary data structures
+        used during typical processing of the pipeline.
+        """
+        self._init_global_nid_offsets()
+        self._init_global_eid_offsets()
+        self._init_ntype_maps()
+        self._init_etype_maps()
+        self._init_ntype_features()
+        self._init_etype_features()
+        self._init_etype_files()
+        self._init_ntype_feature_files()
+        self._init_etype_feature_files()
+
+    def __load_json(self, file_name, base_dir):
+        """Helper function to read the metadata schema file of the input
+        graph dataset. ``file_name`` input argument can be either in absolute
+        or relative path formats. ``base_dir`` is used as the root directory
+        for the input graph dataset where all the files (of the dataset) are
+        expected to be located on the disk.
+
+        Argument
+        --------
+        file_name: string
+            Filename of the input metadata schema file. This can be either
+            in absolute or relative path formats. If in relative format,
+            ``base_dir`` is used as the root directory.
+        base_dir: string
+            Base directory, if provided, is used as the root directory for
+            the input graph dataset
+        """
+        self._basedir = None
+        if base_dir is None:
+            self._basedir = os.getcwd()
+        else:
+            self._basedir = base_dir
+
+        self._metadata_path = os.path.join(
+            os.path.join(self._basedir, file_name)
```

Comment:
What if `file_name` is in absolute path format?

Reply:
The semantics is to provide a file_name, and a base_directory. If these arguments are incorrect or not as expected, then system will through/raise appropriate error. This behavior is the result of earlier comments.